### ISSUE TITLE
Add sitemap.xml

### ISSUE THREAD

**Issue by hemantjadon** [fossasia/loklak_search#452]
**Parent Issue: #311 **

Add `sitemap.xml` for SEO indexing so that users of application can increase.

**Comment by Achint08** [comment]
Yes. @hemantjadon, it would be a good step, how about using [Angular Universal](https://universal.angular.io/) for SEO?

**Comment by hemantjadon** [comment]
@Achint08 we don't have a server so can't use angular universal

**Comment by hemantjadon** [comment]
@mariobehling 
* I was reading about `sitemap.xml` and other SEO techniques, on Google's webmasters [guides](https://www.google.com/webmasters/).
* There I found some interesting information which we can add to our setup.
* As we are running an Angular application on gh-pages, which is a SPA, there I found how Google rates the pages positively and negatively.
* When we create a `sitemap.xml` there are guidelines that our site should be in a canonical form, what this means is If we have for example two URLs `http://example.com` and `http://www.example.com` which serve same content we should always choose one canonical form and use that only in sitemap.
* If we submit multiple urls which serve same content, we get in trouble as it impacts negatively our rating

Now I will come to what is the scenerio in our case as I mentioned in my [blog](https://docs.google.com/document/d/1JoSOKhD2_upWmJATF2J-OOTEPYZgelxmx6UNW9PBmPY/edit) 

* We are hosted on `gh-pages`, which is an awsome service for static pages, but doesn't work well with SPA and dynamic pages.
* The problem is that we are not serving URL's directly via the server
* This compels us to use `dirty spa hack` this is mentioned in my [blog](https://docs.google.com/document/d/1JoSOKhD2_upWmJATF2J-OOTEPYZgelxmx6UNW9PBmPY/edit). 
* This is necessary for the application on gh-pages.
* But when google will try to reach our page  they will land on a 404.html page which redirects to our page.
* I don't know this is a bad practice in SEO but this is why I want some guidance here.
* Is adding redirective pages in sitemap.xml good Idea?

Another thing I will like to mention I would like to mention `http://loklak.net` to google search console if it is not there. This will be a good place to start [console](https://www.google.com/webmasters/tools/home?hl=en). Please visit this page.
It requires to verify the ownership of the page.
Please register to search console and from there we can see what we should do, as we let google tell us what is correct.

**Comment by mariobehling** [comment]
@hemantjadon These are all good points. We can tackle them step by step. Please get started with adding meta information and a simple sitemap. 

> But when google will try to reach our page they will land on a 404.html page which redirects to our page. I don't know this is a bad practice in SEO but this is why I want some guidance here.

Your points are right, but I think a non-perfect solution is better than no solution here. The challenge is we are not on a server. So, please open relevant issues for this as you proceed and we see, who has idea and what is @Orbiter's input.

**Comment by hemantjadon** [comment]
@mariobehling 
>Another thing I would like to mention http://loklak.net to google search console if it is not there. This will be a good place to start [console](https://www.google.com/webmasters/tools/home?hl=en).

Can we do do this?

> So, please open relevant issues for this as you proceed and we see

Okay I will open the relevent issues and start by adding relevent meta tags

**Comment by Orbiter** [comment]
The problem that similar content on different pages causes bad ranking for the whole domain can be solved with the canonical html tag: https://support.google.com/webmasters/answer/139066?hl=en

Pleas implement that tag where appropriate.

**Comment by simsausaurabh** [comment]
Issue captures most of the points discussed here: #311 for SEO, so any one of these issues can be closed.
